go/mobile/async_test.go

```diff
@@ -12,22 +12,31 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-type asyncTestEvent struct{}
-
-func (e *asyncTestEvent) EventHandler(typ string, id, data string, err error) {
-	defer wgAsyncTest.Done()
-	log.Info("Test event received. Id: ", id)
-	_err := ""
-	if err != nil {
-		_err = err.Error()
-	}
-	receivedEvents.Lock()
-	defer receivedEvents.Unlock()
-	receivedEvents.Map[id] = event{
-		Typ:  typ,
-		Id:   id,
-		Data: data,
-		Err:  _err,
+func eventHandler(eventCh chan Event, stopCh chan bool) {
+	for {
+		select {
+		case <-stopCh:
+			log.Info("Stopping check events routine")
+			return
+		case ev := <-eventCh:
+			log.Info("Test event received. Id: ", ev.TicketId)
+			_err := ""
+			if ev.Err != nil {
+				_err = ev.Err.Error()
+			}
+			receivedEvents.Lock()
+			receivedEvents.Map[ev.TicketId] = event{
+				Typ:  ev.Type,
+				Id:   ev.TicketId,
+				Data: ev.Data,
+				Err:  _err,
+			}
+			receivedEvents.Unlock()
+			wgAsyncTest.Done()
+
+		default:
```

Comment:
no need to sleep here, this will only introduce latency.  This select doesn't need a default case.

Reply:
Amended